lambda_functions/delivery_status_processor_lambda/delivery_status_processor_lambda.py

```diff
@@ -34,12 +35,53 @@
     logger.warning("Invalid log level specified, defaulting to INFO")
 
 
+# Duplicated in vetext_incoming_forwarder.
+def validate_twilio_event(event):
+    logger.info("validating twilio delivery event")
+    try:
+        ssm_client = boto3.client("ssm", "us-gov-west-1")
+        auth_ssm_key = os.getenv("TWILIO_AUTH_TOKEN_SSM_NAME", "")
+        if not auth_ssm_key:
+            logger.error("TWILIO_AUTH_TOKEN_SSM_NAME not set")
+            return False
+
+        response = ssm_client.get_parameter(
+            Name=auth_ssm_key,
+            WithDecryption=True
+        )
+        auth_token = response.get("Parameter").get("Value")
+        signature = event["headers"].get("x-twilio-signature", "")
+    except Exception as e:
+        logger.error("SMS retrieval error: %" % e)
+        return False
+
+    if not auth_token or not signature:
+        logger.error("TWILIO_AUTH_TOKEN or signature not set")
+        return False
+    try:
+        validator = RequestValidator(auth_token)
+        uri = "https://%s/vanotify/sms/deliverystatus" % event["headers"]["host"]
+
+        return validator.validate(
+            uri=uri,
+            params=event["body"],
+            signature=signature
+        )
+    except Exception as e:
+        logger.error("Twilio library exception: %" % e)
```

Comment:
should be  `logger.error("Twilio library exception: %s" % e)` to clear the CodeQL issue